### ISSUE TITLE
torch dist error when using DDP in pytorch_submission_base.py

### ISSUE THREAD

**Issue by priyakasimbeg** [mlcommons/algorithmic-efficiency#328]
Pytorch imagenet workloads in target_setting_algorithms are broken with Attribute error from passing in int instead of tensor to a torch dist reduce function. Other workloads here may be broken too. 

## Description

Traceback

```
Traceback (most recent call last):
  File "submission_runner.py", line 599, in <module>
    app.run(main)
  File "/usr/local/lib/python3.8/dist-packages/absl/app.py", line 312, in run
    _run_main(main, args)
  File "/usr/local/lib/python3.8/dist-packages/absl/app.py", line 258, in _run_main
    sys.exit(main(argv))
  File "submission_runner.py", line 572, in main
    score = score_submission_on_workload(workload,
  File "submission_runner.py", line 507, in score_submission_on_workload
    timing, metrics = train_once(workload, global_batch_size,
  File "submission_runner.py", line 320, in train_once
    optimizer_state, model_params, model_state = update_params(
  File "/algorithmic-efficiency/reference_algorithms/target_setting_algorithms/pytorch_submission_base.py", line 61, in update_params
    n_valid_examples = dist_nn.all_reduce(n_valid_examples)
  File "/usr/local/lib/python3.8/dist-packages/torch/distributed/nn/functional.py", line 226, in all_reduce
    return _AllReduce.apply(op, group, tensor)
  File "/usr/local/lib/python3.8/dist-packages/torch/distributed/nn/functional.py", line 439, in forward
    tensor = tensor.clone()
```

## Steps to Reproduce
Run
```
torchrun --redirects 1:0,2:0,3:0,4:0,5:0,6:0,7:0 --standalone --nnodes=1 --nproc_per_node=8 submission_runner.py     --framework=pytorch     --data_dir=/data/imagenet/pytorch     --imagenet_v2_data_dir=/data/imagenet/pytorch     --experiment_dir=/debugging_experiments/     --experiment_name=timing    --workload=imagenet_vit     --submission_path=reference_algorithms/target_setting_algorithms/pytorch_nadamw.py     --tuning_search_space=reference_algorithms/target_setting_algorithms/imagenet_vit/tuning_search_space.json
```
## Source or Possible Fix
Issue is type error `n_valid_examples` should be a tensor instead of int. Working on fix.

**Comment by priyakasimbeg** [comment]
@runame do you have some time to fix this? Seems like casting to torch.Tensor wasn't enough because the num_valid_examples has to be a CUDA tensor.

**Comment by priyakasimbeg** [comment]
Tested this works https://github.com/mlcommons/algorithmic-efficiency/pull/336

**Comment by runame** [comment]
Sorry for the late reply, I missed your question last week. I added a follow-up fix to #325.
